File: fleet_system/fleet_system/report/service_request/service_request.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */


frappe.query_reports["Service Request"] = {
	get_datatable_options(options) {
		return Object.assign(options, {
			checkboxColumn: true,
		});
	},
	onload: function(report) {	
		buttons = document.querySelector('.custom-actions')
		// console.log(buttons)
		
		buttons.classList.remove('hidden-xs')
		buttons.classList.remove('hidden-md')
        // Add a custom button to save the checkbox value
        report.page.add_inner_button(__('Approve'), function() {
			let checked_rows_indexes = report.datatable.rowmanager.getCheckedRows();
			let checked_rows = checked_rows_indexes.map(i => report.data[i]);
			// create empty array which will save name, barcode and itemcode 
			console.log(checked_rows)
			const selected_data = new Array();
			
			// in this loop only specific fields are saved in new arrays NAME, BARCODE, ITEMCODE
			checked_rows.forEach(row=>/* row is an variable inside this loop, in each iteration 
									it's value is dictionary. from this dictionary we are saving required fields*/
				{	// push data as dictionary in new array
					console.log(row.name)

					
					frappe.call('erpnext.fleet_ms.doctype.service_request.service_request.update_approved_status',
					{'status': 'Approved','name': row.name})
					.then(r => {
						alert('Success!');
					});
					// update_approved_status('Approved', row.name);
					console.log(report)
				}	
			)
			frappe.msgprint('Success!');
	
        })
		report.page.add_inner_button(__('Reject'), function() {
            let checked_rows_indexes = report.datatable.rowmanager.getCheckedRows();
			let checked_rows = checked_rows_indexes.map(i => report.data[i]);
			// create empty array which will save name, barcode and itemcode 
			console.log(checked_rows)
			const selected_data = new Array();
			
			// in this loop only specific fields are saved in new arrays NAME, BARCODE, ITEMCODE
			checked_rows.forEach(row=>/* row is an variable inside this loop, in each iteration 
									it's value is dictionary. from this dictionary we are saving required fields*/
				{	// push data as dictionary in new array
					console.log(row.name)
					frappe.call('erpnext.fleet_ms.doctype.service_request.service_request.update_approved_status',{'status': 'Rejected','name': row.name})
					.then(r => {
						alert('Success!');
					});
					console.log(report)
				}	
			)
			frappe.msgprint('Success!');
        });
    },
	
	"filters": [
		{
			"fieldname": "license_plate",
			"label": __("Vehicle"),
			"fieldtype": "Link",
			"options": "Fleet Vehicle"
		}
	]
};

File: fleet_system/fleet_system/report/service_request/service_request.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals

import frappe
from frappe import _
from frappe.utils import flt


def execute(filters=None):
	filters = frappe._dict(filters or {})
	columns = get_columns()
	data = get_data(filters)
	# chart = get_chart_data(data, filters)

	return columns, data, None
	# , chart

def get_columns():
	return [
		{
			'fieldname': 'license_plate',
			'fieldtype': 'Link',
			'label': _('Vehicle'),
			'options': 'Vehicle'
		},
		{
			'fieldname': 'days_left',
			'fieldtype': 'Int',
			'label': 'Days Left'
		},
		{
			'fieldname': 'initiated_date',
			'fieldtype': 'Date',
			'label': _('Initiated Date')
		},
		{
			'fieldname': 'due_date',
			'fieldtype': 'Date',
			'label': _('Due Date')
		},
		{
			'fieldname':'allowed_amount',
			'fieldtype':'Currency',
			'label':'Recommended Amount'
		},
		{
			'fieldname':'service_type',
			'fieldtype':'Data',
			'label':'Item'
		},
		

	]


def get_data(filters):

	conditions, values = get_conditions(filters)

	data = frappe.db.sql("""
		SELECT tsr.license_plate, tsr.name,  DATEDIFF(tsr.due_date, CURDATE()) AS days_left, service_type, tsr.initiated_date, tsr.due_date, tsr.allowed_amount
		FROM `tabService Request` tsr WHERE tsr.status='Recommended' {0}""".format(conditions), values, as_dict=1)
	
	return data


def get_conditions(filters):
	conditions = ''
	values = {}

	if filters.license_plate:
		conditions += ' and tsr.license_plate = %(license_plate)s'
		values['license_plate'] = filters.license_plate

	return conditions, values



@frappe.whitelist()
def update_approved_status(status, name):
    doc = frappe.get_doc("Service Request", name)
    doc.status = status
    doc.save()
    frappe.msgprint("Record updated successfully!")